nodejs/week2/app.js

```diff
@@ -0,0 +1,89 @@
+import express from "express";
+import fs from "fs/promises";
+import path from "path";
+import { fileURLToPath } from "url";
+
+const app = express();
+const port = process.env.PORT || 3000;
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const filePath = path.join(__dirname, "documents.json");
+
+app.use(express.json());
+
+app.get("/search", async (req, res) => {
+  try {
+    const data = await fs.readFile(filePath, "utf-8");
+    const documents = JSON.parse(data);
```

Comment:
These two lines are duplicated in 3 APIs, one idea would be to extract them into a function

```suggestion
    const data = await fs.readFile(filePath, "utf-8");
    const documents = JSON.parse(data);
```